tests/arp/test_wr_arp.py

```diff
@@ -0,0 +1,173 @@
+
+        logger.info('Copying ferret config file to {0}'.format(ptfhost.hostname))
+        ptfhost.copy(src=VXLAN_CONFIG_FILE, dest='/tmp/')
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setupFerret(self, duthost, ptfhost):
+        '''
+        Sets Ferret service on PTF host. This class-scope fixture runs once before test start
+
+        @param duthost: AnsibleHost object of DUT
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        ptfhost.copy(src="scripts/ferret.py", dest="/opt")
+
+        result = duthost.shell(
+            cmd='''ip route show proto zebra type unicast |
+            sed -e '/default/d' -ne '/0\//p' |
+            head -n 1 |
+            sed -ne 's/0\/.*$/1/p'
+            '''
+        )
+        assert len(result['stderr_lines']) == 0, 'Could not obtain DIP'
+
+        dip = result['stdout']
+        logger.info('VxLan Sender {0}'.format(dip))
+
+        ptfhost.host.options['variable_manager'].extra_vars.update({
+            'ferret_args': '-f /tmp/vxlan_decap.json -s {0}'.format(dip)
+        })
+
+        logger.info('Copying ferret config file to {0}'.format(ptfhost.hostname))
+        ptfhost.template(src='ferret.conf.j2', dest='/etc/supervisor/conf.d/ferret.conf')
+
+        logger.info('Generate pem and key files for ssl')
+        ptfhost.command(
+            cmd='''openssl req -new -x509 -keyout test.key -out test.pem -days 365 -nodes 
+            -subj "/C=10/ST=Test/L=Test/O=Test/OU=Test/CN=test.com"''',
+            chdir='/opt'
+        )
+
+        self.__prepareVxlanConfigData(duthost, ptfhost)
+
+        logger.info('Refreshing supervisor control with ferret configuration')
+        ptfhost.shell('supervisorctl reread && supervisorctl update')
+
+    @pytest.fixture(scope='class', autouse=True)
+    def copyPtfDirectory(self, ptfhost):
+        '''
+        Copys PTF directory to PTF host. This class-scope fixture runs once before test start
+
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        ptfhost.copy(src="ptftests", dest="/root")
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setupRouteToPtfhost(self, duthost, ptfhost):
+        '''
+        Sets routes up on DUT to PTF host. This class-scope fixture runs once before test start
+
+        @param duthost: AnsibleHost object of DUT
+        @param ptfhost: AnsibleHost object of PTF
+        '''
+        result = duthost.shell(cmd="ip route show table default | sed -n 's/default //p'")
+        assert len(result['stderr_lines']) == 0, 'Could not find the gateway for management port'
+
+        gwIp = result['stdout']
+        ptfIp = ptfhost.host.options['inventory_manager'].get_host(ptfhost.hostname).vars['ansible_host']
+
+        route = duthost.shell(cmd='ip route get {0}'.format(ptfIp))['stdout']
+        if 'PortChannel' in route:
+            logger.info("Install explicit route through eth0 (mgmt) interface, when we don't have correct route to ptf")
```

Comment:
Really minor nitpick - this reads more like a comment describing what this block is for than a log message. I would probably opt for something a little more specific to this IP like "Installing explicit route through mgmt, no route to ptf for {VIP}".

Reply:
This comment is from wr_arp.yml. it serve for debugging purpose. Will add more details